modules/tensor_mechanics/src/userobjects/StepUserObject.C

```diff
@@ -0,0 +1,96 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "StepUserObject.h"
+#include <limits>
+
+registerMooseObject("TensorMechanicsApp", StepUserObject);
+
+InputParameters
+StepUserObject::validParams()
+{
+  InputParameters params = GeneralUserObject::validParams();
+
+  params.addRequiredParam<std::vector<Real>>(
+      "times",
+      "The times at which the objects are to be enabled/disabled at the beginning of the steps.");
+  params.addRequiredParam<unsigned int>(
+      "number_steps",
+      "The number of loading steps defined for the simulation. The input 'times' "
+      "must have the same size as the number of steps.");
```

Comment:
Any reason for adding this parameter? Why not just deduce it from the number of supplies start times?

Reply:
I can do that. It's not clear to me that it is better to the user. I will modify the times definition to make it more verbose. Hopefully users read it.